libraries/protocol/include/scorum/protocol/scorum_virtual_operations.hpp

```diff
@@ -285,6 +285,52 @@ struct proposal_virtual_operation : public virtual_operation
 
     protocol::proposal_operation proposal_op;
 };
+
+struct allocate_cash_from_advertising_budget_operation : public virtual_operation
+{
+    allocate_cash_from_advertising_budget_operation()
+    {
+    }
+    allocate_cash_from_advertising_budget_operation(const budget_type type_,
+                                                    const account_name_type& owner_,
+                                                    const int64_t id_,
+                                                    const asset& cash_)
+        : type(type_)
+        , owner(owner_)
+        , id(id_)
+        , cash(cash_)
+    {
+        FC_ASSERT(cash.symbol() == SCORUM_SYMBOL);
+    }
+
+    budget_type type = budget_type::post;
+    account_name_type owner = SCORUM_PROXY_TO_SELF_ACCOUNT;
```

Comment:
`SCORUM_PROXY_TO_SELF_ACCOUNT` bad name for empty name. Maybe it is better to keep default ctor for `account_name_type owner`;